lib/geomancer.ex

```diff
@@ -1,21 +1,19 @@
 defmodule Geomancer do
   @moduledoc """
-  A library to convert between various geospatial file formats
+  A behaviour to convert geospatial data from one format to another.
   """
 
-  alias Geomancer.Shapefile
+  @type input_path() :: String.t()
+  @type conversion() :: String.t()
+  @type reason() :: String.t()
+  @type geo_struct() :: struct | map
```

Comment:
I find this type to be confusing. At the very least I would post_fix it with something besides struct since it can also be a map. I'm not sure that it makes sense to type something that can be either a shapefile struct or a geo_json decoded map. They seem too different though I understand you need something to give the callback.